linopy/solvers.py

```diff
@@ -884,17 +890,36 @@ def run_mosek(
                 m.linkfiletostream(mosek.streamtype.log, log_fn, 0)
 
             if warmstart_fn:
-                m.readdata(warmstart_fn)
+                # Gurobi solution format is not supported by MOSEK.
+                # What is the warmstart file? A Gurobi .sol file? The gurobi
+                # docs are a bit sparse on the format. Does it include dual
+                # information?
```

Comment:
This is probably a left-over, can you take that out? 
A warmstart file is a simplex basis which can be used by some solvers as a starting point for the optimization. It is stored as a `.bas` file. However I am not sure whether `xpress` can handle these.  Did you try out the original call?

Reply:
The original call used `task.readdata`. I'll check, but I am reasonably sure that it will only read problem files (LP, MPS, etc). MOSEK does have a .bas file but that is an entirely different format.
But I implemented a `.sol` reader here so I'll, have to fix that too.